core/src/main/scala/kafka/controller/KafkaControllerZkUtils.scala

```diff
@@ -0,0 +1,497 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+package kafka.controller
+
+import java.util.Properties
+
+import kafka.api.LeaderAndIsr
+import kafka.common.TopicAndPartition
+import kafka.log.LogConfig
+import kafka.server.ConfigType
+import kafka.utils.{Logging, ZkUtils}
+import org.apache.zookeeper.KeeperException.Code
+import org.apache.zookeeper.data.Stat
+import org.apache.zookeeper.{CreateMode, KeeperException}
+
+import scala.collection.mutable
+
+class KafkaControllerZkUtils(zookeeperClient: ZookeeperClient, isSecure: Boolean) extends Logging {
```

Comment:
Over time, other components will be using this class, should we name this sth more general that's not tied to controller?

Reply:
I was hoping to do a later "cleanup" PR that would contain:
1. renaming KafkaControllerZkUtils
2. removing state change logger entirely
3. doing your above suggestion of removing the ith tuple element notation ex: x._1